File: src/main/frontend/src/app/models/project.ts
import { ResourceRequirement } from "./resourceRequirement";

export class Project{
    projectId:any;
	clientName:any;
	clientLocation:any;
	state:any;
	projectName:any;
	description:any;
	projectManagerId:any;
	departmentName:any;
	clientId:any;
	clientLocationId:any;
	departmentList:any[] = [];
	allClientLocationList;
	deptId:any;
	active:any;
	syncProject:any;
	createdByName: any;
	empId: any;
	createdOn: any;
	createdBy: any;
	updatedBy:any;
	updatedByName: any;
	teamList: any[] = [];
	isHOD:any;
	projectManagerName:any;
	isTeamCreated:any;
	rejectReason:any;       // RMG project reject reason
	isDraftProject:any;
	bulkSyncList: any[] = [];
	id:any;
	teamsApprovedList:any;
	//Project from Po
	name:any;
	status:any;
	clientState:any;
	department:any;
	projectManager:any;
	poProjectId:any;

	//Team
	teamName:any;
	teamMemberList:any;
	isActive:any;
	teamId:any;
	spoc:any;

	// As per RMG request
	resourceRequirements: ResourceRequirement[];
	oldresourceRequirements: ResourceRequirement[];


	endDate: any;
	isAllProj: any;

	isSynced: any;
	poNo: any;
	startDate:any;
	poStartDate:any;
	poEndDate:any;
	projectType:any;
	poProjectType:any;
	isMail:any;
	apmosysRM:any;
	clientRM:any;
    projectCompletionDate:any;
	projectStatus:any;
	projectManagers:any;
	projectViewId:any;
	projectOverheadId:any;
	projectOverheadName:any;
	internalProjectType:any;
	teamLeadId:any;
	teamLead:any;
	employeeTeamMapId:any;
	draftStatus:any;
  combinedProjectType: string;
  extendedDate: any; // Added for milestone extension
  page:any;
  size:any;
  isClientDashboard:any;
  rescRemovedBy:any;	
}

File: src/main/java/com/apmosys/employeeportal/model/Project.java
package com.apmosys.employeeportal.model;

import java.sql.Timestamp;
import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import org.hibernate.envers.Audited;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;
//import lombok.Getter;
//import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@Audited
@ToString
@Table(name="projects")
public class Project {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer projectId;
	
	private String clientName;
	private String clientLocation;
	private String state;
//	@Column(unique = true)
	private String projectName;
	private String description;
	private Long projectManagerId;
	private Long empId;
	@Column(name="approved_on")
	private Timestamp approvedOn;
    
	private Timestamp createdOn;
	private Integer clientId;
	private String departmentName;
	private Long poProjectId;
	private String active;
	private String syncProject;
	private String isDraftProject;
	private Long createdBy;
	private Long updatedBy;
	private LocalDateTime updatedOn;
	private String role;
    private Integer count;
    private String experience;
    private String poStartDate;
	private String poEndDate;
	private String poNo;
	private String poProjectType;
	private String apmosysRM;
	private String clientRM;
	private Boolean isRenewable;
	private String deptId;
	private String status;
	private String apmosysRmEmail;
	private String projectCompletionDate;
	private String projectStatus;
	private String internalProjectType;
	private Boolean hasClientSideId;
	
}
